src/genie/libs/parser/iosxr/show_mpls.py

```diff
@@ -71,6 +71,144 @@ def cli (self, output=None):
+    
+    """Parser for show mpls ldp discovery"""
+
+    cli_command = ['show mpls ldp discovery']
+
+    def cli(self, output=None):
+        if output is None:
+            out = self.device.execute(self.cli_command)
+        else:
+            out = output
+
+        vrf = "default"
+        # initial return dictionary
+        result_dict = {}
+        discovery_flag = False
+
+        # Local LDP Identifier: 10.52.26.119:0 
+        p1 = re.compile(r'^(VRF +(?P<vrf>\S+):)? *Local +LDP +Identifier: ' 
+                        '(?P<local_ldp_identifier>[\d\.\:]+)$') 
+        
+        # Discovery Sources: 
+        p2 = re.compile(r'^Discovery +Sources:$') 
+        
+        # Bundle-Ether1 : xmit/recv 
+        p3 = re.compile(r'^((?P<interface>\S+) +): *(?P<xmit>xmit)?\/?(?P<recv>recv)?$') 
+        
+        # VRF: 'default' (0x60000000) 
+        p4 = re.compile(r'^VRF: \'(?P<vrf_name>\S+)\' +\(\d+x\d+\)$') 
+        
+        # LDP Id: 10.52.31.244:0, Transport address: 10.52.31.244 
+        p5 = re.compile(r'^(?P<ldp_tdp>\w+) +Id:(?P<space>\s{1,2})?(?P<ldp_tdp_id>[\d\.\:]+),' 
+                        ' +Transport +address: +(?P<transport_ip_addr>[\d\.]+)$') 
+        
+        # Hold time: 15 sec (local:15 sec, peer:15 sec) 
+        p6 = re.compile(r'^Hold +time: +(?P<holdtime_sec>\d+) +sec ' 
+                        '\(local:(?P<proposed_local>\d+) +sec, ' 
+                        'peer:(?P<proposed_peer>\d+) +sec\)$') 
+        
+        # Established: Nov  6 14:39:26.164 (5w2d ago) 
+        p7 = re.compile(r'^Established: (?P<established_date>\S.*) \(\S* ago\)$') 
+
+
+        for line in out.splitlines(): 
+            line = line.strip()
+
+            # Local LDP Identifier: 10.52.26.119:0 
+            m = p1.match(line) 
+            if m:  
+                group = m.groupdict()
+                ldp_dict = result_dict.setdefault('vrf', {}).setdefault(vrf, {}) 
+                local_ldp_identifier_dict = ldp_dict.setdefault('local_ldp_identifier', {}).setdefault(group['local_ldp_identifier'], {}) 
+                continue 
+
+            # Discovery Sources: 
+            m = p2.match(line) 
+            if m:  
+                discovery_flag = True 
+                continue 
+
+            # Bundle-Ether1 : xmit/recv 
+            m = p3.match(line) 
+            if m:  
+                group = m.groupdict() 
+                interface = group['interface'] if group['interface'] else "default" 
+                interface_dict = local_ldp_identifier_dict.setdefault('discovery_sources', {}).setdefault('interfaces', {}).setdefault(interface, {}) 
+                interface_dict.update({'xmit': True if group['xmit'] else False}) 
+                interface_dict.update({'recv': True if group['recv'] else False}) 
+                continue 
+
+            # VRF: 'default' (0x60000000) 
+            m = p4.match(line)
+            if m:
+                group = m.groupdict()
```

Comment:
maybe you can learn the vrf value from this regex?

Reply:
Not sure how to handle this since VRF is defined for each 'interface' and not globally, as [iosxe](https://github.com/CiscoTestAutomation/genieparser/blob/master/src/genie/libs/parser/iosxe/tests/ShowMplsLdpDiscovery/cli/equal/golden_output_all_output.txt).

Should I change the schema from:

```
schema = {
        'vrf': {
            Any(): {
                Optional('local_ldp_identifier'): {
                    Any(): {
                        Optional('discovery_sources'): {
                            'interfaces': {
                                Any(): {
                                    Optional('source_ip_addr'): str,
...
```

To 

```
schema = {
            'local_ldp_identifier': {
                Any(): {
                    Optional('discovery_sources'): {
                        'interfaces': {
                            Any(): {
                                Optional('vrf'): str,
...
```

Reply:
yes, from the device output, I think this schema is better

Reply:
ok it's patched.

Reply:
ah, @RomainGuilloux sorry for the inconvenience, if we already have the schema for the same command on `iosxe`, then it's better to keep the same structure for `iosxr`. Could you please use the previous schema.

Reply:
Fixed

Reply:
confirmed above change.